desc/objectives/getters.py

```diff
@@ -102,6 +105,9 @@ def get_fixed_axis_constraints(eq, profiles=True, normalize=True):
                 constraints += (
                     con(eq=eq, normalize=normalize, normalize_target=normalize),
                 )
+    for param in ["I", "G", "Phi_mn"]:
+        if np.array(getattr(eq, param, [])).size:
+            constraints += (FixParameter(eq, param),)
```

Comment:
It would probably be worth creating a new linear objective function like `FixSheetCurrent` or something that fixes these three parameters. Like how we have other fixed objectives that are redundant with `FixedParameter` for common use cases. 

But also you can do this with a single objective function as `FixParameter(eq, ["I", "G", "Phi_mn"])` instead of creating 3 separate constraints. And instead of the for loop with `gettattr`, you should be able to check `if {"I", "G", "Phi_mn"} <= set(eq.optimizable_params):`. If that doesn't work we need to add these sheet current variables to the equilibrium optimizable params.

Reply:
same for other instances below

Reply:
> It would probably be worth creating a new linear objective function like `FixSheetCurrent` or something that fixes these three parameters. Like how we have other fixed objectives that are redundant with `FixedParameter` for common use cases.

I think that can be another PR. If the sheet current terms exist it's because the user explicitly included them which means they're usually not fixed. This is mainly just to handle the case of the fixed boundary solve inside the proximal optimizer, or other cases where you're doing a "fixed boundary" like solve with the output of a free boundary solve.

> But also you can do this with a single objective function as `FixParameter(eq, ["I", "G", "Phi_mn"])` instead of creating 3 separate constraints. And instead of the for loop with `gettattr`, you should be able to check `if {"I", "G", "Phi_mn"} <= set(eq.optimizable_params):`. If that doesn't work we need to add these sheet current variables to the equilibrium optimizable params.

`{"I", "G", "Phi_mn"} <= set(eq.optimizable_params)` doesn't do what we want. They are always listed under `optimiizable_params` but if no sheet current is assigned they're just empty arrays of length 0, in which case we don't need to fix them (similar to how we don't include constraints for profiles that don't exist, even though the attributes are still there and return an empty array)